rust/src/builtins/scriptlet_intercept/groupadd.rs

```diff
@@ -0,0 +1,150 @@
+//! CLI handler for intercepted `groupadd`.
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use cap_std::fs::Dir;
+use clap::{Arg, Command};
+use std::io::Write;
+
+/// Entrypoint for (the rpm-ostree implementation of) `groupadd`.
+pub(crate) fn entrypoint(args: &[&str]) -> Result<()> {
+    fail::fail_point!("intercept_groupadd_ok", |_| Ok(()));
+
+    // This parses the same CLI surface as the real `groupadd`,
+    // but in the end we only extract the group name and (if
+    // present) the static GID.
+    let matches = cli_cmd().get_matches_from(args);
+    let gid: Option<u32> = match matches.value_of("gid") {
+        None => None,
+        Some(s) => Some(s.parse()?),
+    };
+    let groupname = matches
+        .value_of("groupname")
+        .expect("missing required groupname");
```

Comment:
Shouldn't dump core on this:

```suggestion
        .ok_or_else(|| anyhow!("missing required groupname"))?;
```